substrate/frame/conviction-voting/src/traits.rs

```diff
@@ -0,0 +1,40 @@
+use crate::AccountVote;
+use frame_support::dispatch::DispatchResult;
+
+pub trait VotingHooks<AccountId, Index, Balance> {
+	// Called when vote is executed.
+	fn on_vote(who: &AccountId, ref_index: Index, vote: AccountVote<Balance>) -> DispatchResult;
+
+	// Called when removed vote is executed.
+	// is_finished indicates the state of the referendum = None if referendum is cancelled, Some(true) if referendum is ongoing and Some(false) when finished.
+	fn on_remove_vote(who: &AccountId, ref_index: Index, ongoing: Option<bool>);
```

Comment:
`ongoing` should be an enum that expresses the state better.

Reply:
@bkchr, in `conviction_voting` runtime implementation, I see that the type `Polls` is defined as `Referenda`. is it possible through `Polls` in the trait `Config` to bring the enum `ReferendumInfo` in `conviction_voting`  and use it here? or do we necessarily need to add the corresponding type in the runtime implementation & the trait config?

Reply:
I don't really understand your plan. Why not just make it an enum as I said above?

Reply:
If the enum items are the same I just thought it would be nice to re-use what we already have in pallet-referenda.

Reply:
The pallets are not tightly coupled and I also don't see a reason why we should do this here now. Let's try some custom enum for now :)